api/blueprints/drbCollection.py

```diff
@@ -89,6 +92,74 @@ def collectionCreate(user=None):
 
     return APIUtils.formatOPDS2Object(201, opdsFeed)
 
+@collection.route('/update/<uuid>', methods=['POST'])
+@validateToken
+def collectionUpdate(uuid, user=None):
+    logger.info('Handling collection update request')
+
+    collectionData = request.json
+    dataKeys = collectionData.keys()
+
+    if {'title', 'creator', 'description'}.issubset(set(dataKeys)) == False\
+            or len(set(dataKeys) & set(['workUUIDs', 'editionIDs'])) == 0:
+        errMsg = {
+            'message':
+                'title, creator and description fields are required'
+                ', with one of workUUIDs or editionIDs to create a collection'
+        }
+
+        return APIUtils.formatResponseObject(400, 'createCollection', errMsg)
+
+    dbClient = DBClient(current_app.config['DB_CLIENT'])
+    dbClient.createSession()
+
+    #Getting the collection the user wants to replace
+    try:
+        collection = dbClient.fetchSingleCollection(uuid)
+    except NoResultFound:
+        errMsg = {'message': 'Unable to locate collection {}'.format(uuid)}
+        return APIUtils.formatResponseObject(404, 'fetchSingleCollection', errMsg)
+
+    workUUIDs=collectionData.get('workUUIDs', [])
+    editionIDs=collectionData.get('editionIDs', [])
+
+    collection.title = collectionData['title']
+    collection.creator = collectionData['creator']
+    collection.description = collectionData['description']
+
+    #Deleting the rows of collection_editions that were in the original collection
+    dbClient.session.execute(COLLECTION_EDITIONS.delete().where(COLLECTION_EDITIONS.c.collection_id == collection.id))
+
+    #Inserting rows of collection_editions based on workUUIDs/editionIDs arrays
+    if len(editionIDs) > 0:
+        for editID in editionIDs:
+            dbClient.session.execute(COLLECTION_EDITIONS.insert().values \
+                (collection_id = collection.id, edition_id = editID))
```

Comment:
generally more pythonic to do `if editionIDs`, but no big deal.

Other than that, performance wise, it would be better to do one insert instead of separate inserts in a loop, ie:
```
dbClient.session.execute(COLLECTION_EDITIONS.insert().values([
    {"collection_id": collection.id, "edition_id": eid}
    for eid in editionIDs
])

Reply:
Nice advice on cleaning up the code and making it more efficient.